### PR TITLE
Ruff format

### LINKED ISSUE #1187: Use ruff format instead of black
We use ruff anyways so we can drop one dependency.

### PR DESCRIPTION
**Pull request by jstucke**
- updated ruff version and rules
- replaced black with ruff
- checked, fixed and reformatted code with new ruff version

resolves #1187

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -49,19 +44,25 @@ markers = [
...
     # pydantic only supports these from python>=3.9
     "UP006",
     "UP007",
+    # rules may cause conflicts when used with the formatter
+    "ISC001",
+    "Q001",
 ]
-fixable = ["F", "E", "W", "C90", "N", "UP", "B", "A", "PL", "C4", "EXE", "ISC", "PIE", "T20", "PT", "RET", "SIM", "TCH",
-    "PTH", "PLR", "PLW", "ARG", "ERA", "RUF", "Q"]
+fixable = ["F", "E", "W", "C90", "N", "UP", "B", "A", "C4", "EXE", "FA", "ISC", "PIE", "T20", "PT", "Q", "RET", "SIM",
```
Comment (maringuu) [inline pyproject.toml:64]: While we are at it, can we replace this by `ALL`?

File: pyproject.toml [hunk 2]
```diff
@@ -49,19 +44,25 @@ markers = [
 ]
 
 [tool.ruff]
-select = ["F", "E", "W", "C90", "N", "UP", "B", "A", "PL", "C4", "EXE", "ISC", "PIE", "T20", "PT", "RET", "SIM", "TCH",
-    "PTH", "PLR", "PLW", "ARG", "ERA", "RUF", "Q"]
+select = ["F", "E", "W", "C90", "N", "UP", "B", "A", "C4", "EXE", "FA", "ISC", "PIE", "T20", "PT", "Q", "RET", "SIM",
```
Comment (maringuu) [inline pyproject.toml:47]: Could you convert this to a sorted list with one line per entry?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -43,13 +43,9 @@ repos:
...
     -   id: jshint
 
--   repo: https://github.com/ambv/black
-    rev: 23.3.0
-    hooks:
-    -   id: black
-
 - repo: https://github.com/charliermarsh/ruff-pre-commit
-  rev: 'v0.0.275'
+  rev: 'v0.1.11'
```
Comment (maringuu) [inline .pre-commit-config.yaml:47]: Since it took me so long to review this, we can bump the version once again.

File: pyproject.toml [hunk 4]
```diff
@@ -1,8 +1,3 @@
-[tool.black]
-line-length = 120
-skip-string-normalization = true
-target-version = ['py38']
-
 [tool.pylint.main]
```
Comment (maringuu) [inline pyproject.toml:1]: While not strictly related to this PR, can you remove this?

**Review by maringuu — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (jstucke) [reply pyproject.toml:47]: I don't know if sorting it alphabetically is actually helpful: Right now it is ordered the same way as the rules in [the Ruff docs](https://docs.astral.sh/ruff/rules/) which makes it easier to see changes in new ruff versions

File: pyproject.toml (hunk 4 above)
Comment (jstucke) [reply pyproject.toml]: I tried to merge it with the ruff configuration where possible and removed the rest

**Comment by maringuu** [comment]
Do not wonder, I prettified the history a bit :)

**Review by maringuu — APPROVED** [review]

**Pull request merged**